documentation/coding_guidelines.md

```diff
@@ -227,6 +227,18 @@ impl Foo {
 }
 ```
 
+### Integer Arithmetic
+
+As every integer operation (`+`, `-`, `/`, `*`, etc.) implies edge-cases (e.g. overflows `u64::MAX + 1`, underflows `0u64 -1`, division by zero, etc.),
+we use checked arithmetic instead of directly using math symbols.
+It forces us to think of edge-cases, and handle them explicitely.
+This is a brief and simplified mini guide of the different functions that exist to handle integer arithmetic:
+
+* [checked_](https://doc.rust-lang.org/std/primitive.u32.html#method.checked_add): use this function if you want to handle overflows and underflows as a special edge-case. It returns `None` if an underflow or overflow has happened, and `Some(operation_result)` otherwise.
+* [overflowing_](https://doc.rust-lang.org/std/primitive.u32.html#method.overflowing_add): use this function if you want the result of an overflow to potentially wrap around (e.g. `u64::MAX.overflow_add(1) == 0`). It returns the underflowed or overflowed result as well as a flag indicating if an overflow has occured or not.
```

Comment:
I think it'll be more clear to give the example `u64::MAX.overflow_add(10 == (0, true)`, so for someone lazy as me who skims through, when reading "except that it returns the result DIRECTLY" doesn't get confused.

Reply:
oops, good point!